src/core/lib/surface/call.c

```diff
@@ -989,12 +989,15 @@ static grpc_error *consolidate_batch_errors(batch_control *bctl) {
   if (n == 0) {
     return GRPC_ERROR_NONE;
   } else if (n == 1) {
-    return bctl->errors[0];
+    grpc_error *e = bctl->errors[0];
+    bctl->errors[0] = NULL;
```

Comment:
why is the n == 1 branch a special case and no ERROR_REFERENCING is created in this case?